### PR TITLE
create test of main window

### PR DESCRIPTION
**Pull request by AntonenkoTymofii**

### CODE REVIEW

File: src/test/java/org/example/MainWindowTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class MainWindowTest {
+
+    private MainWindow mainWindow;
```
Comment (sonakrrr) [inline src/test/java/org/example/MainWindowTest.java]: You better change it to `MainWindowTest mainWindow`

**Review by sonakrrr — CHANGES_REQUESTED** [review]

File: src/test/java/org/example/MainWindowTest.java (hunk 1 above)
Comment (AntonenkoTymofii) [reply src/test/java/org/example/MainWindowTest.java]: I think u want to say MainWindow mainWindowTest

Comment (sonakrrr) [reply src/test/java/org/example/MainWindowTest.java]: Oh, u r absolutely right!

File: src/test/java/org/example/MainWindowTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        assertEquals(650, mainWindow.getHeight());
+        assertEquals(JFrame.EXIT_ON_CLOSE, mainWindow.getDefaultCloseOperation());
+    }
+
+    @Test
+    void testOpenTableAction() {
+        ActionEvent e = mock(ActionEvent.class);
+        JButton openTable = (JButton) mainWindow.getJMenuBar().getComponent(1);
+        openTable.getActionListeners()[0].actionPerformed(e);
+        verify(table, times(1)).setVisible(true);
```
Comment (sonakrrr) [inline src/test/java/org/example/MainWindowTest.java]: In this line you need to replace `1` with `0`

**Review by sonakrrr — CHANGES_REQUESTED** [review]

File: src/test/java/org/example/MainWindowTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    MainWindow mainWindowTest;
+    MyTable table;
+
+    @BeforeEach
+    void setUp() {
+        MyTable.setInstance(table);
+        table = mock(MyTable.class);
+        mainWindowTest = new MainWindow();
+    }
```
Comment (sonakrrr) [inline src/test/java/org/example/MainWindowTest.java:23]: I think u need to use headless system, as I see the reason of error in CI

**Review by sonakrrr — CHANGES_REQUESTED** [review]

File: src/test/java/org/example/MainWindowTest.java (hunk 3 above)
Comment (AntonenkoTymofii) [reply src/test/java/org/example/MainWindowTest.java:23]: Idk but maybe it will help.. :/

Comment (AntonenkoTymofii) [reply src/test/java/org/example/MainWindowTest.java:23]: > I think u need to use headless system, as I see the reason of error in CI



> I think u need to use headless system, as I see the reason of error in CI



> I think u need to use headless system, as I see the reason of error in CI

nothing changed.....

**Comment by AntonenkoTymofii** [comment]
I have no idea how to testing GUI in OS without GUI. So I think it doesn't make sense...

**Comment by AntonenkoTymofii** [comment]
I have no idea how to testing GUI in OS without GUI. So I think it doesn't make sense...

**Pull request closed without merging**